warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryPruningVisitor.java

```diff
@@ -516,10 +540,28 @@ private void replaceAndAssign(JexlNode toReplace, String queryString, JexlNode b
             JexlNode parent = toReplace.jjtGetParent();
             if (parent != null && parent != toReplace) {
                 if (queryString != null && log.isDebugEnabled()) {
-                    log.debug("Pruning " + queryString + " to " + (baseReplacement instanceof ASTTrueNode ? "true" : "false"));
+                    if (this.debugPrune && baseReplacement != null) {
+                        log.debug("Pruning " + queryString + " to " + (baseReplacement instanceof ASTTrueNode ? "true" : "false"));
+                    }
                 }
                 
-                JexlNodes.swap(parent, toReplace, baseReplacement);
+                if (baseReplacement != null) {
+                    JexlNodes.swap(parent, toReplace, baseReplacement);
+                } else {
+                    // remove the node entirely
+                    List<JexlNode> children = new ArrayList<>(parent.jjtGetNumChildren() - 1);
+                    for (int i = 0; i < parent.jjtGetNumChildren(); i++) {
+                        JexlNode child = parent.jjtGetChild(i);
+                        if (!child.equals(toReplace)) {
```

Comment:
should/can this be (child != toReplace) instead of (!child.equals(toReplace))

Reply:
yep it should be the exact object so we can use !=